studio/tests.py

```diff
@@ -53,6 +58,31 @@ def test_permissions_are_removed(self):
         self.assertFalse(has_perm)
 
 
+@pytest.mark.django_db
+def test_do_delete_account():
+    """
+    Test function helpers.do_delete_account()
+    """
+    test_user = {"username": "foo@test.com", "email": "foo@test.com", "password": "bar"}
+    user = User.objects.create_user(test_user["username"], test_user["email"], test_user["password"])
+    assert user is not None and user.id > 0
+    assert user.is_active is True
+
+    profile = UserProfile(user=user, is_approved=True)
+    profile.save()
+    assert profile.deleted_on is None
+
+    user_id = user.id
+    user_account_deleted = do_delete_account(user_id)
+
+    assert user_account_deleted is True
+
+    user = User.objects.get(pk=user_id)
+
+    assert user.is_active is False
+    assert user.userprofile.deleted_on >= datetime.now(timezone.utc) - timedelta(seconds=1)
```

Comment:
I guess this could theoretically fail if there is a slow database connection right?

Reply:
Yes. I will increase to 10 secs.